x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/sp/CloudServiceProvider.java

```diff
@@ -6,30 +6,40 @@
 
 package org.elasticsearch.xpack.idp.saml.sp;
 
+import org.elasticsearch.common.Nullable;
 import org.elasticsearch.common.Strings;
+import org.elasticsearch.xpack.idp.privileges.ServiceProviderPrivileges;
 import org.joda.time.Duration;
 import org.joda.time.ReadableDuration;
+import org.opensaml.security.x509.X509Credential;
 
-import java.net.URI;
-import java.net.URISyntaxException;
 
 public class CloudServiceProvider implements SamlServiceProvider {
 
     private final String entityid;
-    private final URI assertionConsumerService;
+    private final String assertionConsumerService;
     private final ReadableDuration authnExpiry;
+    private final String nameIdPolicyFormat;
+    private final X509Credential signingCredential;
+    private final ServiceProviderPrivileges privileges;
+    private final boolean signAuthnRequests;
+    private final boolean signLogoutRequests;
 
-    public CloudServiceProvider(String entityId, String assertionConsumerService) {
+    public CloudServiceProvider(String entityId, String assertionConsumerService, String nameIdPolicyFormat,
+                                ServiceProviderPrivileges privileges, boolean signAuthnRequests, boolean signLogoutRequests,
+                                @Nullable X509Credential signingCredential) {
         if (Strings.isNullOrEmpty(entityId)) {
             throw new IllegalArgumentException("Service Provider Entity ID cannot be null or empty");
         }
         this.entityid = entityId;
-        try {
-            this.assertionConsumerService = new URI(assertionConsumerService);
-        } catch (URISyntaxException e) {
-            throw new IllegalArgumentException("Invalid URI for Assertion Consumer Service", e);
-        }
+        this.assertionConsumerService = assertionConsumerService;
```

Comment:
I still feel like this should be a URI. It _is_ a URI and I think we should consistently treat it like one.

Reply:
I don't necessarily agree that this is important.

I reverted the change as you requested. There is value in validating that it is a URL (it's a URI but we should not be validating for a URI, as it can't be a URN ) and we will be doing this in the Registration API , no harm in doing it here too. Again, I think that for all intents and purposes when it comes to SAML, handling this as a URL in code - after validation - is unnecessary and it can potentially  introduce subtle bugs. 
For instance we need to take care so that when comparing `ACS` values , we are comparing string representations of the URLs and i.e. _not_ [`URL.equals`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URL.html#equals(java.lang.Object)) which does something different. 

I didn't change the IDP configuration settings to be `Setting<URL>` too because of [the arguments here](https://github.com/elastic/elasticsearch/pull/51682#discussion_r373320101), the fact that we already validate them and finally that we have no good existing way to parse and handle `Setting<URL>` with optional values in a way that makes sense to me ( I can further elaborate if needed )

Reply:
I'm moving these to Setting<URL> in the IDP metadata PR